config/config.exs

```diff
@@ -0,0 +1,5 @@
+use Mix.Config
```

Comment:
We should avoid config in libraries (https://hexdocs.pm/elixir/library-guidelines.html#anti-patterns). The way we use it here seems to provide defaults (for parameterised types, and for `use UUID.Ecto.Type`) so that's probably borderline OK :) but I still wouldn't do it, if users of the library want to have defaults in app env, I'd make them do that themselves.

Reply:
@wojtekmach yeah, my problem was I wanted to test that config values worked and that they were overridden by passing the `using` macro opts, and also acted as a fallback for missing opts.

However, they are added at compile time (in `TypeConfig.compile_type_config/2`) so when I tried using `Application.put_env` in the ExUnit`setup` macro, they weren't applied. 😢

Reply:
I think that looking through the public interface angle can help here too, e.g. 

```elixir
assert TestUUID3.generate() == UUID.uuid3(:dns, "mydomain.com")
```

(which you have already) is good enough to test that passing the options works correctly, if it didn't there's no way this would be true!